src/routes/safe/components/Transactions/TxList/BatchExecute.tsx

```diff
@@ -0,0 +1,240 @@
+import Button from 'src/components/layout/Button'
+import { ButtonStatus, Modal } from 'src/components/Modal'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+import { Transaction } from 'src/logic/safe/store/models/types/gateway.d'
+import { fetchSafeTransaction } from 'src/logic/safe/transactions/api/fetchSafeTransaction'
+import { generateSignaturesFromTxConfirmations } from 'src/logic/safe/safeTxSigner'
+import { getExecutionTransaction } from 'src/logic/safe/transactions'
+import { getGnosisSafeInstanceAt, getMultisendContractAddress } from 'src/logic/contracts/safeContracts'
+import { EMPTY_DATA } from 'src/logic/wallets/ethTransactions'
+import { getMultiSendJoinedTxs, MultiSendTx } from 'src/logic/safe/transactions/multisend'
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { getBatchableTransactions } from 'src/logic/safe/store/selectors/gatewayTransactions'
+import { Dispatch } from 'src/logic/safe/store/actions/types'
+import { ModalHeader } from 'src/routes/safe/components/Balances/SendModal/screens/ModalHeader'
+import Row from 'src/components/layout/Row'
+import Paragraph from 'src/components/layout/Paragraph'
+import { GnosisSafe } from 'src/types/contracts/gnosis_safe'
+import Hairline from 'src/components/layout/Hairline'
+import { getInteractionTitle } from 'src/routes/safe/components/Transactions/helpers/utils'
+import PrefixedEthHashInfo from 'src/components/PrefixedEthHashInfo'
+import { getExplorerInfo } from 'src/config'
+import { fetchTransactionDetails } from 'src/logic/safe/store/actions/fetchTransactionDetails'
+import { createMultiSendTransaction } from 'src/logic/safe/store/actions/TxMultiSender'
+import { BatchExecuteHoverContext } from 'src/routes/safe/components/Transactions/TxList/BatchExecuteHoverProvider'
+import { TxArgs } from 'src/logic/safe/store/models/types/transaction'
+import { isTxPending } from 'src/logic/safe/store/selectors/pendingTransactions'
+import {
+  getTxConfirmations,
+  getTxInfo,
+  getTxRecipient,
+  getTxValue,
+} from 'src/routes/safe/components/Transactions/TxList/utils'
+import { TX_LIST_EVENTS } from 'src/utils/events/txList'
+import { DecodedTxDetailType } from 'src/routes/safe/components/Apps/components/ConfirmTxModal'
+import { DecodeTxs } from 'src/components/DecodeTxs'
+import { DecodedDataParameterValue } from '@gnosis.pm/safe-react-gateway-sdk'
+import { trackEvent } from 'src/utils/googleTagManager'
+
+async function getBatchExecuteData(
+  dispatch: Dispatch,
+  transactions: Transaction[],
+  safeInstance: GnosisSafe,
+  safeAddress: string,
+  account: string,
+) {
+  const batchableTransactionsWithDetails = await Promise.all(
+    transactions.map(async (transaction) => {
+      if (transaction.txDetails) return transaction
+
+      const txDetails = await dispatch(fetchTransactionDetails({ transactionId: transaction.id }))
+
+      const transactionDetails = txDetails || (await fetchSafeTransaction(transaction.id))
```

Comment:
You don't need to fallback here. `fetchTransactionDetails` fetches the transaction if it doesn't find it in the store.

Reply:
Currently, `fetchTransactionDetails` can return undefined if the transaction in question already has txDetails or there is no `safeAddress`.

Reply:
I think we should modify the thunk. If the `txDetails` are there, they should be returned.

Reply:
I adjusted `fetchTransactionDetails` to return them if they already exist